### DISCUSSION TITLE
Make commands.dat editable for Unturned

### DISCUSSION THREAD

**Discussion by medic4085** [gameserverapp/Platform#531] [Suggestions]
_**Explination**_
As of now.  When an unturned server is created.  There is "extremely limited customization" that can be made to the server.  Let me try and explain.  

What works.  
1.  Workshop maps/mods.  Essentially, anything via mods works because it involves using the WorkshopDownloadConfig.json file, which does not get overwritten via server boot.
2. Rocketmod items.  I haven't completely tested, but nothing appears to be affected from the time of me writing this. 

What does not work.  
1.  You list a basic config.json file.  This is great, however, different modes that are listed are explicitly controlled via the commands.dat file.  For example, anything defined in the easy, normal or hard config.  Is defined by the line {Mode "Easy,Normal,Hard"} in the commands.dat file.  Being that you can't customize/add that line.  You essentially cannot make config changes.  Or tell the server which to use.  I am assuming it is using "normal" by default.  So easy and hard cannot be defined.   
2.  **Partial Correction**. See the correction at the end of this comment.  Your current commands.dat file contains "Cheats Enabled".  However, it only allows cheats to "owners" or "admins", which neither can be defined.  So essentially, no one can use them.  The "owner" is defined in the commands.dat file.  While "admins" are defined in the adminlist.dat.  Both files can be edited, however, upon server start, are overwritten to default values. - __**Correction**__ Owner is defined in the commands.dat file.  Once in-game, the owner can run the command /admin steamid within the chat/console and it will add them as admins to the adminlist.  So what I found is once the owner is defined, they can add admin/whitelist/blacklist from the commands within game and it will populate in the other files within /saves/server.  Don't believe anything other than the commands.dat file is editable outside the game.
3.  Custom Loadouts.  Again, unable to define it due to the commands.dat file being overwritten.  
4. Further testing proves that even the config settings are not properly being referenced.  For example, anything under "Browser" portion is not working.  I have tested this on another server to confirm that my .json file was valid.  It works on there.  

Here is a list of most, not all possible commands for the commands.dat file.    

map washington/yukon/PEI/custom
port 27015
name
perspective both/third/first
cheats enabled/disabled
chatrate
maxplayers
playermax
welcome message
mode easy/normal/hard
loadout 0/
loadout 1/
loadout 2/
loadout 3/
loadout 4/
loadout 5/        I believe(not 100%) there is a limit of 10 custom loadouts.
loadout 6/
loadout 7/
loadout 8/
loadout 9/
loadout 10/
pvp/pve
hide_admins
timeout 1000
owner steamid

_**To Reproduce**_
1. Modify commands.dat or any file within /saves/server
4. Upon server start, all files within /saves/server are overwritten to default entries
5. This essentially nullifies any changes or customization to those files and limits config file entries.

_**Expected behavior**_
Changes made to the commands.dat would not be overwritten.

_**Screenshots**_
I Could provide it, but I don't think it is necessary unless requested.

_**Additional context**_
I understand or at least think that certain parameters are defined in the commands.dat file via the dashboard/config.  Let me explain.

name - defined via "overview-Server List Name".
Map - defined via "overview-map"
maxplayers - defined via "overview-slots"
port - defined via"danger zone-advanced-gameports"
GSLT - defined via"config-template-game-Game Server Login Token (GLST)
cheats enabled - Unknown

This all being said, if we could get additional parameters within config sections, preferably below the Game Server Login Token it could solve most if not all of these problems.  

I encourage anyone to reply to this with any information that may help explain.  Or possibly supply an easier fix to this problem.

**Comment by shinji257** [comment]
Alternatively allow for a user managed custom file (example: `commands_custom.dat` or `commands.custom`) where an individual may add additional command that is appended to the commands.dat file before the server boots.  This is under the assumption that the commands.dat file is rebuilt with each server boot.  This would allow for the user to add additional commands that may not necessarily be supported by the current server manager implementation.

**Comment by birdman3131** [comment]
Note. I don't currently have a linux server (Or the game) to check but at one point the admin list for ark was moved to the config template under settings. It might be there for unturned as well,

![image](https://github.com/gameserverapp/Platform/assets/749937/c24609d3-9873-449c-be21-78526513c1cf)

**Comment by mvaessen** [comment]
Hi, Thanks for your feedback!

I converted the Issue into a Suggestion as it's not about a broken feature but missing functionality. Development items and breaking issues can be posted in Issues. I understand missing features can be breaking for the experience, but it's not the same as a broken feature. :)

I'm happy to look into making the option to customise the `commands.dat` file after ASA launched

**Comment by medic4085** [comment]
I also want to state that the game is not referencing the default config correctly.  For example, any changes made to the default config or custom configs within the config templates are not being committed to the server on boot.  I'm not sure if it has anything to do with how it is referenced in that section "server.cfg" instead of config.json.  

I at least tested the config.json on another platform and it displays the icon's, thumbnails, descriptions and links within the games server browser.  However, when loaded in GSA it only populates the server name.  

Wanted to make this a point as it is technically a separate issue from the commands.dat issue.
